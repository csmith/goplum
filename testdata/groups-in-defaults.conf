defaults {
  interval = 30s
  timeout = 20s
  groups = ["webservices", "databases"]
}

group "webservices" {
  alert_limit = 3
  alert_window = 10m
}

group "databases" {
  alert_limit = 2
  alert_window = 5m
}

alert debug.sysout "test" {}

check debug.random "test-check" {
  # This check should inherit groups from defaults
}