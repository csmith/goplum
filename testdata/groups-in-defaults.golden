{
  "Alerts": {
    "test": {}
  },
  "Checks": {
    "test-check": {
      "Name": "test-check",
      "Type": "debug.random",
      "Config": {
        "Alerts": [
          "*"
        ],
        "Groups": [
          "webservices",
          "databases"
        ],
        "Interval": 30000000000,
        "Timeout": 45000000000,
        "GoodThreshold": 2,
        "FailingThreshold": 2
      },
      "Check": {
        "PercentGood": 0.5
      },
      "LastRun": "0001-01-01T00:00:00Z",
      "Scheduled": false,
      "Settled": false,
      "State": "indeterminate",
      "Suspended": false,
      "History": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
      ]
    }
  },
  "Groups": {
    "databases": {
      "Name": "databases",
      "AlertLimit": 2,
      "AlertWindow": 300000000000,
      "Defaults": null
    },
    "webservices": {
      "Name": "webservices",
      "AlertLimit": 3,
      "AlertWindow": 600000000000,
      "Defaults": {
        "Alerts": null,
        "Groups": null,
        "Interval": 0,
        "Timeout": 45000000000,
        "GoodThreshold": 0,
        "FailingThreshold": 0
      }
    }
  }
}